backend/routes/factions.go

```diff
@@ -202,3 +204,68 @@ func getFactions(w http.ResponseWriter, r *http.Request) {
 	}
 	routeutils.WriteDataJson(w, string(factions))
 }
+
+func getFactionMembers(w http.ResponseWriter, r *http.Request) {
+	factionID, err := strconv.Atoi(r.URL.Query().Get("factionID"))
+	if err != nil || factionID < 0 {
+		routeutils.WriteErrorJson(w, http.StatusBadRequest, "Invalid faction ID")
+		return
+	}
+
+	pageLength, err := strconv.Atoi(r.URL.Query().Get("pageLength"))
+	if err != nil || pageLength <= 0 {
+		pageLength = 10
+	}
+	if pageLength > 50 {
+		pageLength = 50
+	}
+
+	page, err := strconv.Atoi(r.URL.Query().Get("page"))
+	if err != nil || page <= 0 {
+		page = 1
+	}
+	offset := (page - 1) * pageLength
+
+	query := `
+		SELECT user_address, allocation
+		FROM FactionMembersInfo
+		WHERE faction_id = $1
+		ORDER BY allocation DESC
+		LIMIT $2 OFFSET $3
+	`
+
+	rows, err := core.ArtPeaceBackend.Databases.Postgres.Query(context.Background(), query, factionID, pageLength, offset)
```

Comment:
Lines 237-268 can be cleaned up, if you use the `PostgresQueryJson` with a custom faction members type, like this example: https://github.com/keep-starknet-strange/art-peace/blob/5f5aa850349a1e83ac1d632ea8d4a3e6e80f00fd/backend/routes/factions.go#L161